### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by jsm13**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -60,16 +60,25 @@ In short, the notion of which variables are available where.
...
 
 #### `let`
 
-So far, the primary way to control scope in an application has been through the use
-of functions:
+The fact that other blocks (`if`, `for`, `while`, etc) *do not* create scopes is one of the oddities for which JS does and should get knocked.
+
+ES6 introduces the `let` keyword which works just like `var` but is scoped to its block (`{...}`) rather than its function.
+
+> Objects use curly braces `{}` but are not code blocks!
```
Comment (jsm13) [inline readme.md]: "Object literal definitions" better maybe?

File: readme.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ Each version contains features / changes to be added to the language.
 
 In short, I like to think of ECMAScript as the language, and JavaScript an implementation of that language.
 
-#### Evolution of JS
+### Evolution of JS
 
 > Check out [this awesome visualization](http://shaunlebron.github.io/solar-system-of-js/#0) of the current state of the JS universe
```
Comment (jsm13) [inline readme.md:28]: This is wonderful. I plan to go through it in class

File: readme.md [hunk 3]
```diff
@@ -135,41 +142,66 @@ var a = 2;
 
 ### Default parameters (5 / 40)
 
-With ES6, we now have the option add set a default value for any of our functions' parameters.
+If a function defined to take 3 parameters is called with two arguments, what is the value of the third parameter in the function body?
```
Comment (jsm13) [inline readme.md]: I think "if a function defined with 3 parameters..." is more accurate. This is just an opportunity to distinguish argument and parameter

File: readme.md [hunk 4]
```diff
@@ -135,41 +142,66 @@ var a = 2;
...
+
+For example:
+
+```js
+function printAndSum(a, b, c){
+console.log(a)
+console.log(b)
+console.log(c)
```
Comment (jsm13) [inline readme.md]: Need to fix indentation

File: readme.md [hunk 5]
```diff
@@ -197,9 +229,9 @@ function greetUser ({ name, location })  {
 1. https://github.com/ga-wdi-exercises/es6-exercises/blob/master/06-deconstruction.js
 
 
-## Break (15 / 90)
+## Break (10 / 90)
 
-### Concise Object Properties and Methods (5 / 95)
+### Concise Object Literal Definitions (Properties and Methods) (5 / 95)
```
Comment (jsm13) [inline readme.md:239]: this was fine as is. Can remove the change

File: readme.md [hunk 6]
```diff
@@ -362,8 +401,8 @@ let add = (x,y) => (
 Support for ES6 is great! - https://kangax.github.io/compat-table/es6/
 
 If you need to support a legacy browser, check out the following tools:
-- [Traceur](https://github.com/google/traceur-compiler/wiki/Getting-Started)
 - [Babel](https://babeljs.io/)
+- [Traceur](https://github.com/google/traceur-compiler/wiki/Getting-Started)
```
Comment (jsm13) [inline readme.md:424]: I think Babel is significantly more popular and the likelihood of a student reading up on both isn't great

File: readme.md (hunk 1 above)
Comment (juancgarcia) [reply readme.md]: Maybe this? `All the object literals that we've created so far use curly braces`...

File: readme.md (hunk 3 above)
Comment (juancgarcia) [reply readme.md]: Maybe this is an opportunity to be verbose for the sake of clarity.

`We've defined a function below with 3 parameters. If we invoke the function and only supply 2 arguments,`...

File: readme.md [hunk 7]
```diff
@@ -135,41 +142,66 @@ var a = 2;
...
+
+printAndSum(1, 2)
+
+// => ???
+```
+
+With ES6, we now have the option to set a default value for any of our functions' parameters.
 
 ```js
 function hello( name = "stranger"){
```
Comment (juancgarcia) [inline readme.md]: In the interest of continuity I think this ES6 example function should be an updated version of the prior ES5 function. Either by promoting `hello` to an ES5 version above (and altering the preamble text accordingly), or replacing `hello` here with the ES6 "parameter defaults" version of `printAndSum`

File: readme.md [hunk 8]
```diff
@@ -236,13 +269,15 @@ let y = 2
 let obj = {x,y}
 ```
 
+NOTE: this is not `let` giving us this ability. `vars` and `conts` can be used to set object properties the same way in es6
```
Comment (juancgarcia) [inline readme.md]: Typo: `const` vs `conts`

File: readme.md [hunk 9]
```diff
@@ -282,15 +320,15 @@ foods.forEach(function(food){
 })
 ```
 
-If there is more than one argument to the anonymous function, wrap
+If there is more or fewer than one argument to the anonymous function, wrap
```
Comment (juancgarcia) [inline readme.md]: This line might be confusing. Consider the following:


`Just like in ES5 we wrap our arguments in parens.`

(original foods.forEach example)

`However, when an ES6 function has only one parameter defined, we're allowed to omit the parens`

(supply single param function example, something like the following)

`So the following function,`

```js
let myLogger = (message) => console.log(`Best console log ever!: ${message}`)
```

`can also be written like this.
`
```js
let myLogger = message => console.log(`Best console log ever!: ${message}`)
```

File: readme.md (hunk 7 above)
Comment (jsm13) [reply readme.md]: 👍 will use `printAndSum`

File: readme.md (hunk 8 above)
Comment (jsm13) [reply readme.md]: 😳 will fix

File: readme.md (hunk 9 above)
Comment (jsm13) [reply readme.md]: 👍

File: readme.md (hunk 1 above)
Comment (jsm13) [reply readme.md]: Was worried that implied there is some other form of object literal. Went with
```
> We write object literals with curly braces but these are not code blocks.
> Object literals should always follow an assignment or be provided as an argument to a function but never stand on their own.
> Code blocks should be used as parts of special forms like for function bodies or the bodies of the flow control forms listed above.
```

**Pull request merged**